core/src/main/java/org/jdbi/v3/BeanMapper.java

```diff
@@ -51,6 +67,58 @@ public BeanMapper(Class<T> type)
         }
     }
 
+    /**
+     * Maps a property name to possible column names candidates. This default implementation will map the property name
+     * as-is, and converted from the camel-case syntax to an underscore separated name.
+     * <p>
+     * Processing of the returned mappings will be case insensitive. The mappings are not expected to change across
+     * invocations.
+     *
+     * @param propertyName the bean property name
+     *
+     * @return a collection holding the possible column names (mutable in the default implementation)
+     */
+    protected Collection<String> mapToColumnNames(String propertyName)
+    {
+        Locale locale = getLocale();
+
+        List<String> columnName = new ArrayList<>(5);
```

Comment:
Actually, I prefer to be slightly oversized regarding what subclasses may want to do with that list...